packages/react-reconciler/src/ReactFiberCommitWork.old.js

```diff
@@ -536,7 +537,16 @@ function commitHookEffectListUnmount(
             }
           }
 
-          safelyCallDestroy(finishedWork, nearestMountedAncestor, destroy);
+          if (__DEV__ && (flags & HookInsertion) !== NoHookEffect) {
+            setIsRunningInsertionEffect(true);
+            try {
```

Comment:
Although this logic could move to `safelyCallDestroy` which has its own try/catch, I keep it here for symmetry.